libs/core/src/lib/step-input/step-input.component.ts

```diff
@@ -0,0 +1,555 @@
+    @Input()
+    incrementButtonIcon: string = 'add';
+
+    /** Sets icon displayed in Decrement Button */
+    @Input()
+    decrementButtonIcon: string = 'less';
+
+    /** Sets Increment Button title attribute */
+    @Input()
+    incrementButtonTitle: string = '';
+
+    /** Sets Decrement Button title attribute */
+    @Input()
+    decrementButtonTitle: string = '';
+
+    /** Sets input aria-label attribute */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Sets input id */
+    @Input()
+    inputId: string = `fd-step-input-${stepInputUniqueId++}`;
+
+    /** Sets control value */
+    @Input()
+    value: number = 0;
+
+    /** Sets minimum value boundary */
+    @Input()
+    min: number;
+
+    /** Sets maximum value boundary */
+    @Input()
+    max: number;
+
+    /** Sets input step value */
+    @Input()
+    step: number = 1;
+
+    /** Sets input name attribute */
+    @Input()
+    name: boolean = null;
+
+    /** Sets input title attribute */
+    @Input()
+    inputTitle: string = '';
+
+    /** Sets formatting mode */
+    @Input()
+    mode: 'decimal' | 'currency' = 'decimal';
+
+    /** Sets state of the control. Can be `success`, `error`, `warning`, `information` or blank for default.*/
+    @Input()
+    state: FormStates;
+
+    /** Custom unit displayed as a label next to the input */
+    @Input()
+    unit: string;
+
+    /** Whether to use grouping when formatting value */
+    @Input()
+    useGrouping: boolean = true;
+
+    /** Defines minimal number of fractional digits for control value */
+    @Input()
+    minFractionDigits: number;
+
+    /** Defines maximal number of fractional digits for control value */
+    @Input()
+    maxFractionDigits: number;
+
+    /** Currency used to format value. Check Intl.NumberFormat documentation for more information*/
+    @Input()
+    currency: string;
+
+    /** Currency used to format value. Check Intl.NumberFormat documentation for more information */
+    @Input()
+    currencyDisplay: string;
+
+    /** Whether StepInput should display Increase/Decrease buttons */
+    @Input()
+    hasStepButtons: boolean = true;
+
+    /** Horizontally aligns value inside input */
+    @Input()
+    textAlign: 'left' | 'center' | 'right';
+
+    /** Emits event when input gets focused */
+    @Output()
+    onFocusIn: EventEmitter<void> = new EventEmitter<void>();
+
+    /** Emits event when input loses focus */
+    @Output()
+    onFocusOut: EventEmitter<void> = new EventEmitter<void>();
+
+    /** Emits new value when control value has changed */
+    @Output()
+    valueChange: EventEmitter<number> = new EventEmitter<number>();
+
+    /** @hidden */
+    @ViewChild('incrementBtn', {read: ElementRef})
+    incrementButton: ElementRef;
+
+    /** @hidden */
+    @ViewChild('decrementBtn', {read: ElementRef})
+    decrementButton: ElementRef;
+
+    /** @hidden */
+    @ViewChild('inputElement', {read: ElementRef, static: true})
```

Comment:
You need this reference in ngOnInit?

Reply:
Yes, this is because `writeValue` is called before view initialization and this method calls `_updateViewValue` which requires reference to HTML `input` element.